SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestMultiServerUnrealComponents/SpatialTestMultiServerUnrealComponents.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestMultiServerUnrealComponents.h"
+
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKSettings.h"
+#include "TestUnrealComponents.h"
+#include "TestUnrealComponentsActor.h"
+
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+
+/**
+ * This test tests that static and dynamic component are properly resolved on expectant workers.
+ *
+ * The test includes 2 Server and 2 Clients.
+ * The flow is as follows:
+ * - Setup:
+ *  - The TestActor is spawned with both static and dynamic components.
+ *  - The components are initialized with unique starting values.
+ *  - The TestActor is possessed by Client1
+ * - Test:
+ *	- The owner client checks it can see the actor, components, and the values are correct.
+ *  - The non-auth server checks it can see the actor, components, and the values are correct.
+ *  - The non-owning client checks it can see the actor, components, and the values are correct.
+ * - Clean-up:
+ *	- The TestActor is destroyed.
+ */
+ASpatialTestMultiServerUnrealComponents::ASpatialTestMultiServerUnrealComponents()
+	: Super()
+{
+	Author = "Mike";
+	Description = TEXT("Test Unreal Component Replication in a MultiServer Context");
+}
+
+void ASpatialTestMultiServerUnrealComponents::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, TestActor);
```

Comment:
`ThisClass` what is this sorcery?

Reply:
@improbable-dmitrii put me onto it 🧙

Reply:
Yes, you can see it ObjectMacros.h - similar to "Super".